src/encoded/tests/test_metadata.py

```diff
@@ -0,0 +1,1034 @@
+    ]
+    for param in mr._get_field_params():
+        assert param in expected_field_params
+
+
+def test_metadata_metadata_report_get_at_id_params(dummy_request):
+    from encoded.reports.metadata import MetadataReport
+    dummy_request.environ['QUERY_STRING'] = (
+        'type=Experiment&files.file_type=bigWig&files.file_type=bam'
+        '&files.replicate.library.size_range=50-100'
+        '&files.status!=archived&files.biological_replicates=2'
+    )
+    dummy_request.json = {'elements': ['/experiments/ENCSR123ABC/']}
+    mr = MetadataReport(dummy_request)
+    mr._initialize_at_id_param()
+    mr._maybe_add_json_elements_to_param_list()
+    assert mr._get_at_id_params() == [('@id', '/experiments/ENCSR123ABC/')]
+
+
+def test_metadata_metadata_report_get_default_params(dummy_request):
+    from encoded.reports.metadata import MetadataReport
+    dummy_request.environ['QUERY_STRING'] = (
+        'type=Experiment&files.file_type=bigWig&files.file_type=bam'
+        '&files.replicate.library.size_range=50-100'
+        '&files.status!=archived&files.biological_replicates=2'
+    )
+    dummy_request.json = {'elements': ['/experiments/ENCSR123ABC/']}
+    mr = MetadataReport(dummy_request)
+    assert mr._get_default_params() == [
+        ('field', 'audit'),
+        ('field', 'files.@id'),
+        ('limit', 'all')
+    ]
+
+def test_metadata_metadata_report_build_query_string(dummy_request):
+    from encoded.reports.metadata import MetadataReport
+    dummy_request.environ['QUERY_STRING'] = (
+        'type=Experiment&files.file_type=bigWig&files.file_type=bam'
+        '&files.replicate.library.size_range=50-100'
+        '&files.status!=archived&files.biological_replicates=2'
+    )
+    mr = MetadataReport(dummy_request)
+    mr._build_query_string()
+    mr.query_string == (
+        'type=Experiment&files.file_type=bigWig'
+        '&files.file_type=bam&files.replicate.library.size_range=50-100'
+        '&files.status%21=archived&files.biological_replicates=2'
+        '&field=audit&field=files.%40id&limit=all'
+    )
+
+
+def test_metadata_metadata_report_get_search_path(dummy_request):
+    from encoded.reports.metadata import MetadataReport
+    dummy_request.environ['QUERY_STRING'] = (
+        'type=Experiment&files.file_type=bigWig&files.file_type=bam'
+        '&files.replicate.library.size_range=50-100'
+        '&files.status!=archived&files.biological_replicates=2'
+    )
+    mr = MetadataReport(dummy_request)
+    assert mr._get_search_path() == '/search/'
+
+
+def test_metadata_metadata_report_validate_request(dummy_request):
+    from encoded.reports.metadata import MetadataReport
+    dummy_request.environ['QUERY_STRING'] = (
+        'type=Experiment&files.file_type=bigWig&files.file_type=bam'
+        '&files.replicate.library.size_range=50-100'
+        '&files.status!=archived&files.biological_replicates=2'
+    )
+    mr = MetadataReport(dummy_request)
+    assert mr._validate_request() is None
```

Comment:
This is a "passed" validation?  It's not very clear

Reply:
Updated to return True. This validation actually redundant now with the @allowed_types decorator. But you could add some other validation logic.